File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/utils/TriState.java
package me.lokka30.levelledmobs.bukkit.utils;

import org.jetbrains.annotations.Nullable;

public enum TriState {
    TRUE,
    UNKNOWN,
    FALSE;

    public static TriState of(final boolean bool) {
        return bool ? TRUE : FALSE;
    }

    public static TriState of(final @Nullable Boolean bool) {
        return bool == null ? UNKNOWN : of(bool);
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/me/lokka30/levelledmobs/bukkit/utils/TriState.kt
package me.lokka30.levelledmobs.bukkit.utils

enum class TriState {
    TRUE,
    UNKNOWN,
    FALSE;

    companion object {
        fun of(boolean: Boolean): TriState {
            return if(boolean) { TRUE } else { FALSE }
        }

        fun of(boolean: Boolean?): TriState {
            return if(boolean == null) {
                UNKNOWN
            } else if(boolean) {
                TRUE
            } else {
                FALSE
            }
        }
    }
}